docs/Lifecycling.md

```diff
@@ -0,0 +1,148 @@
+# Lifecycling
+
+This page aims to explain and clarify
+
+- the goal of the lifecycling functionality in the library
+
+- the assumptions that have been made in its implementation
+
+We will be using dates in the examples below, but the same applies when the time type parameter is `Time`.
+
+## Introduction: evolution of a contract
+
+In the model outlined in [[1]](#1), a financial contract evolves over time. As an example, let us consider the contract `c1`
+
+```Haskell
+When (t ≥ t_1) (Scale (Observe "USD_LIBOR_3M") (When (t ≥ t_2) (One “USD”)))
+```
+
+which we acquired at a time `t_0` with `t_0` ≤ `t_1` ≤ `t_2`.
+
+Economically, this contract observes a certain rate at time `t_1` and pays its amount in dollars at time `t_2`.
+
+In order to understand how the contract evolves over time, we need to recall that `when cond c` acquires the underlying contract on the first instant the condition becomes `True`.
```

Comment:
The use of `cond` here can be confusing (as we have a constructor called `cond`)

Reply:
Thanks, I will clarify this. I also plan to use capitalised names when referring to the tree building blocks, such as `When` or `Cond`